modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -365,7 +371,11 @@ protected HttpMessage createMessage(String[] initialLine) throws Exception {
                     );
             }
             // combines the HTTP message pieces into a single full HTTP request (with headers and body)
-            final HttpObjectAggregator aggregator = new Netty4HttpAggregator(handlingSettings.maxContentLength());
+            final HttpObjectAggregator aggregator = new Netty4HttpAggregator(
+                handlingSettings.maxContentLength(),
+                httpPreRequest -> enabled.get() == false
```

Comment:
I am not entirely sure of the importance of this being in sync with the enabled flag checked in the rest-bulk-handler. However, it seems they could flip at different times, since we register these independently and they each add a listener for cluster settings. I wonder if we can fix that. I realize it is only an issue when turning it on/off, but would be nice to fix anyway.

Reply:
I added a comment to the feature flag ticket and will address it as a follow-up. I agree with the concern.

https://elasticco.atlassian.net/browse/ES-9262